src/services/test_results/TestResultsBuilder.ts

```diff
@@ -1,64 +1,144 @@
-import {TestResultTestRunAttributes} from '../../dto/test_results/TestResultTestRunAttributes';
+import {TestResultUnitTestRunAttributes} from '../../dto/test_results/TestResultUnitTestRunAttributes';
 import {Framework, TestFieldNames, TestRunResults} from '../../dto/test_results/TestResultEnums';
 import {TestResultErrorAttributes} from '../../dto/test_results/TestResultErrorAttributes';
 import {TestResultBuildAttributes} from '../../dto/test_results/TestResultBuildAttributes';
 import {TestResultTestFieldAttributes} from '../../dto/test_results/TestResultTestFieldAttributes';
 import {TestResult} from '../../dto/test_results/TestResult';
 import {TestResultTestField} from '../../dto/test_results/TestResultTestField';
-import {TestResultTestRunElement} from '../../dto/test_results/TestResultTestRunElement';
+import {TestResultUnitTestRunElement} from '../../dto/test_results/TestResultUnitTestRunElement';
 import {TestResultError} from '../../dto/test_results/TestResultError';
 import {WebApi} from 'azure-devops-node-api';
 import * as ba from 'azure-devops-node-api/BuildApi';
 import * as ta from 'azure-devops-node-api/TestApi';
 import {TestCaseResult} from "azure-devops-node-api/interfaces/TestInterfaces";
 import {LogUtils} from "../../LogUtils";
+import {TestResultGherkinTestRunElement} from "../../dto/test_results/TestResultGherkinTestRunElement";
+import {TestResultGherkinTestRunAttributes} from "../../dto/test_results/TestResultGherkinTestRunAttributes";
+import {GherkinTestResultData} from "../../dto/test_results/GherkinTestResultData";
 
 let convert = require('xml-js');
 let xmlescape = require('xml-escape');
+const globby = require('globby');
+const {readFileSync} = require('fs-extra');
 
 export class TestResultsBuilder {
+    private static readonly CUCUMBER_TEST_TYPE = "Cucumber";
 
-    public static buildTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
+    public static buildUnitTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
         if (!testResults || !testResults.length) {
-            logger.info('No test results were retrieved/found');
+            logger.info('No unit test results were retrieved/found');
             return null;
         }
-        let testResultTestRuns = this.buildTestResultTestRun(testResults, logger);
+        let testResultTestRuns = this.buildUnitTestResultTestRun(testResults, logger);
```

Comment:
Please think of a better naming, regardless of what originally this was, if you actually changed it.